### ISSUE TITLE
Multiple oauth security schemes not supported

### ISSUE THREAD

**Issue by cbornet** [springfox/springfox#959]
It seems if several OAuth security schemes are defined for the docket, only the last one is mapped to the security definitions

```java
@Bean
  public Docket petApi() {
    return new Docket(DocumentationType.SWAGGER_2)
        .select()
          .apis(RequestHandlerSelectors.any())
          .paths(PathSelectors.any())
          .build()
        .securitySchemes(newArrayList(accessCodeAuth(), implicitAuth()))
   }

private OAuth accessCodeAuth() {
        List<GrantType> grants = newArrayList(
                new AuthorizationCodeGrant(
                        new TokenRequestEndpoint("tre:uri", "treClient", "tre"),
                        new TokenEndpoint("te:uri", "treToken")));
        List<AuthorizationScope> scopes = newArrayList();
        return new OAuth("code_oauth", newArrayList(scopes), newArrayList(grants));
    }

    private OAuth implicitAuth() {
        List<GrantType> grants = newArrayList(
                new ImplicitGrant(
                        new LoginEndpoint("implicit:uri"),
                        "implicitToken"));
        List<AuthorizationScope> scopes = newArrayList();
        return new OAuth("implicit_oauth", newArrayList(scopes), newArrayList(grants));
    }
```
outputs
```json
"securityDefinitions":
    {
        "implicit_oauth":
        {
            "type": "oauth2",
            "authorizationUrl": "implicit:uri",
            "flow": "implicit"
        }
    },
```

**Comment by dilipkrish** [comment]
Would you be willing to try your hand at submitting a PR.

**Comment by cbornet** [comment]
I will have a look. I think this is because in swagger 1.2 the grant types were grouped in one oauth authorization whereas now there is one security reference per grant/flow

**Comment by cbornet** [comment]
It comes from ResourceListing::initializeSecuritySchemes() which transforms List<SecurityScheme> to LinkedHashMap<String, SecurityScheme>. Note that since it is shared with swagger12, changing it again to a list will remove the protection against multiple oauth schemes.

**Comment by ChinmayanP** [comment]
@dilipkrish is this issue fixed in Version 2.9.2 ?
